Sources/NIO/Bootstrap.swift

```diff
@@ -705,3 +705,114 @@ public final class DatagramBootstrap {
         }
     }
 }
+
+/// A `NIOPipeBootstrap` is an easy way to bootstrap a `PipeChannel` which uses two (uni-directional) UNIX pipes
+/// and makes a `Channel` out of them.
+///
+/// Example bootstrapping a `Channel` using `stdin` and `stdout`:
+///
+///     let channel = try NIOPipeBootstrap(group: group)
+///                       .channelInitializer { channel in
+///                           channel.pipeline.addHandler(MyChannelHandler())
+///                       }
+///                       .withPipes(inputDescriptor: STDIN_FILENO, outputDescriptor: STDOUT_FILENO)
+///
+public final class NIOPipeBootstrap {
+    private let group: EventLoopGroup
+    private var channelInitializer: ((Channel) -> EventLoopFuture<Void>)?
+    @usableFromInline
+    internal var _channelOptions = ChannelOptions.Storage()
+
+    /// Create a `PipeBootstrap` on the `EventLoopGroup` `group`.
+    ///
+    /// - parameters:
+    ///     - group: The `EventLoopGroup` to use.
+    public init(group: EventLoopGroup) {
+        self.group = group
+    }
+
+    /// Initialize the connected `PipeChannel` with `initializer`. The most common task in initializer is to add
+    /// `ChannelHandler`s to the `ChannelPipeline`.
+    ///
+    /// The connected `Channel` will operate on `ByteBuffer` as inbound and outbound messages. Please note that
+    /// `IOData.fileRegion` is _not_ supported for `PipeChannel`s because `sendfile` only works on sockets.
+    ///
+    /// - parameters:
+    ///     - handler: A closure that initializes the provided `Channel`.
+    public func channelInitializer(_ handler: @escaping (Channel) -> EventLoopFuture<Void>) -> Self {
+        self.channelInitializer = handler
+        return self
+    }
+
+    /// Specifies a `ChannelOption` to be applied to the `PipeChannel`.
+    ///
+    /// - parameters:
+    ///     - option: The option to be applied.
+    ///     - value: The value for the option.
+    @inlinable
+    public func channelOption<Option: ChannelOption>(_ option: Option, value: Option.Value) -> Self {
+        self._channelOptions.append(key: option, value: value)
+        return self
+    }
+
+    private func validateFileDescriptorIsNotAFile(_ descriptor: CInt) throws {
+        var s: stat = .init()
+        try withUnsafeMutablePointer(to: &s) { ptr in
+            try Posix.fstat(descriptor: descriptor, outStat: ptr)
```

Comment:
@Lukasa this is currently run on the EventLoop. I think I might actually do what you suggested and for now just assert that it's not run on the EventLoop. We can always improve later by allowing that by offloading it.